WordPress/gutenberg
# Button : deprecate `isSmall` prop

Closes #59733

## What?

This PR officially deprecates the `isSmall` prop of the `Button` component.

## Why?

#51842 introduced `size` prop, which covers `isSmall` prop. #53560 also replaced the `isSmall` prop with the `size` prop for `Button` components throughout the Guenberg project. This should allow us to officially deprecate the `isSmall` prop.

## How?

~~I added an error indicating that a deprecation warning appears from **WP6.6** and will be removed in **WP6.9**.~~ At the same time, I updated types and unit tests.

## Testing Instructions

All CIs should be green. Additionally, use the code below to confirm that a console error is displayed when a `Button` component with the `isSmall` prop is registered as a custom block.

```javascript
wp.blocks.registerBlockType( 'test/button-with-is-small', {
	apiVersion: 3,
	title: 'Button with deprecated isSmall prop',
	edit () {
		return wp.element.createElement(
			'div',
			wp.blockEditor. useBlockProps(),
			wp.element.createElement(
				wp.components.Button,
				{ isSmall: true },
				'Push Me',
			),
		);
	},
} );
```

![deprecated-warning](https://github.com/WordPress/gutenberg/assets/54422211/564764ed-a33c-44f3-917b-be36bf59e6ab)
